src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -53,11 +61,28 @@ public List<Route> routes() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
-
-        String workflowId = request.param(WORKFLOW_ID);
-        Template template = Template.parse(request.content().utf8ToString());
-        WorkflowRequest workflowRequest = new WorkflowRequest(workflowId, template);
-        return channel -> client.execute(CreateWorkflowAction.INSTANCE, workflowRequest, new RestToXContentListener<>(channel));
+        try {
+            String workflowId = request.param(WORKFLOW_ID);
+            Template template = Template.parse(request.content().utf8ToString());
+            WorkflowRequest workflowRequest = new WorkflowRequest(workflowId, template);
+            return channel -> client.execute(CreateWorkflowAction.INSTANCE, workflowRequest, ActionListener.wrap(response -> {
+                XContentBuilder builder = response.toXContent(channel.newBuilder(), ToXContent.EMPTY_PARAMS);
+                channel.sendResponse(new BytesRestResponse(RestStatus.CREATED, builder));
+            }, exception -> {
+                try {
+                    FlowFrameworkException ex = (FlowFrameworkException) exception;
+                    XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
+                    channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
+                } catch (IOException e) {
+                    logger.error("Failed to send back create workflow exception", e);
```

Comment:
If we can't build the create workflow exception, what will be returned to the user?

Reply:
Are we using toXContent rather then just returning the specific message we write when we do `listener.onFailure(new FlowFrameworkException(exception.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));`

Reply:
This is similar to how AD handles exceptional requests ([ref](https://github.com/opensearch-project/anomaly-detection/blob/4c6ba48bf9fb9234ad8bc0b9193f3c68409acfb9/src/main/java/org/opensearch/ad/rest/RestDeleteAnomalyResultsAction.java#L81)).

We'll only face this issue if the `FlowFrameworkException` cant be converted to XContent, which probably wont happen.

Reply:
> Are we using toXContent rather then just returning the specific message we write when we do listener.onFailure(new FlowFrameworkException(exception.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));

`FlowFrameworkException::toXContent` just allows us to control the format of the exception message when we write it back as a rest response : 

`builder.startObject().field("error", "Request failed with exception: [" + this.getMessage() + "]").endObject();`

Reply:
yeah just meant since we don't do the xContent thing in AD, would we just return the xContent Parsing exception if that ever happens, I know it will be unlikely